src/v/config/configuration.cc

```diff
@@ -1912,6 +1912,34 @@ configuration::configuration()
       "Retention in bytes for topics created during automated recovery",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       1_GiB)
+  , cloud_storage_recovery_topic_validation_mode(
+      *this,
+      "cloud_storage_recovery_topic_validation_mode",
+      "Validation mode performed before recovering a topic from cloud storage",
+      {.needs_restart = needs_restart::no,
+       .example = "check_manifest_existance",
+       .visibility = visibility::tunable},
+      model::recovery_validation_mode::check_manifest_existence,
+      {model::recovery_validation_mode::check_manifest_existence,
+       model::recovery_validation_mode::check_manifest_own_metadata_only,
+       model::recovery_validation_mode::no_check})
+  , cloud_storage_recovery_topic_validation_depth(
+      *this,
+      "cloud_storage_recovery_topic_validation_depth",
+      "Depth of metadata validation, when "
+      "`cloud_storage_recovery_topic_validation_mode` is "
+      "`check_manifest_own_metadata_only`",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      10)
+  , cloud_storage_recovery_topic_force_override_cfg(
+      *this,
+      "cloud_storage_recovery_topic_force_override_cfg",
+      "If True, force validation to ignore the recovery_checks topic property, "
+      "and instead use cluster config values "
+      "`cloud_storage_recovery_topic_validation_mode` and "
+      "`cloud_storage_recovery_topic_validation_depth`",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      false)
```

Comment:
I'm curious what the motivation is here? Presumably, an admin would have control over the incoming properties, no?

Reply:
an escape hatch to disable validation altogether (cloud_storage_recovery_topic_validation_mode=no_check)

or to help testing to force a validation mode